test/sanity/globalConstants.wtest

```diff
@@ -0,0 +1,103 @@
+const x = 1 + 1
+test "global constants are referenceable" {
+  assert.equals(2, x)
+}
+
+
+const a = b
+const b = 1 + 1
+const c = b
+test "global constants referencing other constants defined before/after are properly initialized" {
+  assert.equals(2, a)
+  assert.equals(2, b)
+  assert.equals(2, c)
+}
+
+const o = p
+object p {
+  const property e = self.f()
+  const property f = o
+  const property g = q
+  const property h = o.itself()
+  const property i = q.itself()
+  const property j = o.itselfThroughConst()
+  const property k = q.itselfThroughConst()
+  const property l = o.itselfThroughOtherConst()
+  const property m = q.itselfThroughOtherConst()
+  method itself() = self
+  method itselfThroughConst() = o
+  method itselfThroughOtherConst() = q
+}
+const q = p
+test "global constants referencing singletons defined before/after are properly initialized" {
+  assert.equals(p, o)
+  assert.equals(p, q)
+  assert.equals(p, p.e())
+  assert.equals(p, p.f())
+  assert.equals(p, p.g())
+  assert.equals(p, p.h())
+  assert.equals(p, p.i())
+  assert.equals(p, p.j())
+  assert.equals(p, p.k())
+  assert.equals(p, p.l())
+  assert.equals(p, p.m())
+}
+
+
+class Person {
+  const property friend
+  const property name
+  method itself() = self
+}
+const jack = new Person(name = "Jack", friend = jill)
+const jill = new Person(name = "Jill", friend = jack)
```

Comment:
This is a hard case, but I think it's the most useful...

To do this, const should be referenceable within it's own initialization. If we could assign an instance to them as we can do with singletons there would be no problem, but we don't know what module that instance would belong to. What if it's a number, which have to be unique in the system?
Maybe we can use a partially initialized runtime object and somehow merge that with the final response?

Reply:
This is THE case to be solved.